app/policies/api/tests_policy.rb

```diff
@@ -0,0 +1,26 @@
+class Api::TestsPolicy < ApplicationPolicy
+  def update?
+    if user.is_a?(User) && record.user_id == user.id
+      true
+    else
+      raise Pundit::NotAuthorizedError, "User does not have permission to access the resource."
+    end
+  end
```

Comment:
The `Api::TestsPolicy` is assuming that only the authorized user can create a new test. Ensure that this assumption is correct and that the appropriate permissions are being checked.